traffic_ops/traffic_ops_golang/federations/ds.go

```diff
@@ -123,11 +131,93 @@ func (v *TOFedDSes) ParamColumns() map[string]dbhelpers.WhereColumnInfo {
 	}
 }
 func (v *TOFedDSes) GetType() string {
-	return "federation_deliveryservice"
+	return "federation deliveryservice"
+}
+
+func (v *TOFedDSes) SetKeys(keys map[string]interface{}) {
+	i, _ := keys["id"].(int)
+	v.fedID = &i
+}
+
+func (v *TOFedDSes) GetKeys() (map[string]interface{}, bool) {
+	if v.fedID == nil {
+		return map[string]interface{}{"id": 0}, false
+	}
+	return map[string]interface{}{"id": *v.fedID}, true
+}
+
+func (v *TOFedDSes) GetAuditName() string {
+	if v.XMLID != nil {
+		return *v.XMLID
+	}
+	return strconv.Itoa(*v.ID)
+}
+
+func (v *TOFedDSes) GetKeyFieldsInfo() []api.KeyFieldInfo {
+	return []api.KeyFieldInfo{
+		{"id", api.GetIntKey},
+	}
 }
 
 func (v *TOFedDSes) Read() ([]interface{}, error, error, int) { return api.GenericRead(v) }
 
+func (v *TOFedDSes) Delete() (error, error, int) {
+	dsIDStr, ok := v.APIInfo().Params["dsID"]
+	if !ok {
+		return errors.New("dsID must be specified for deletion"), nil, http.StatusBadRequest
+	}
+	dsID, err := strconv.Atoi(dsIDStr)
+	if err != nil {
+		return errors.New("dsID must be an integer"), nil, http.StatusBadRequest
+	}
+	v.ID = &dsID
+
+	// Check that we can delete it
+	if respCode, usrErr, sysErr := checkFedDSDeletion(v.APIInfo().Tx.Tx, *v.fedID, dsID); usrErr != nil || sysErr != nil {
+		if usrErr != nil {
+			return usrErr, sysErr, respCode
+		}
+		return usrErr, sysErr, respCode
+	}
+
+	// Actually delete the DS from the Federation
+	if err := deleteFedDS(v.APIInfo().Tx.Tx, *v.fedID, dsID); err != nil {
```

Comment:
Should this check tenancy? I'm sure Perl didn't, so it's probably not strictly necessary for a rewrite, but something to consider.

Reply:
given that I did not add it on the GET rewrite (and the POST that was rewritten and merged before that)  I say that we do not add it here as we should add it to all three at the same time instead of partial support.

Going forward we should come to an agreeance if rewrites should include tenancy checks or not.